packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1598,12 +1602,17 @@ function commitWork(current: Fiber | null, finishedWork: Fiber): void {
             commitHookEffectListUnmount(
               HookLayout | HookHasEffect,
               finishedWork,
+              finishedWork.return,
```

Comment:
It's fine to pass `fiber.return` here because none of these calls get made within the context of an unmounted/unmounting tree.